torchqdynamics/mesolve/mesolve.py

```diff
@@ -131,36 +127,33 @@ def mesolve(
         solver = Dopri45()
 
     # define the QSolver
-    args = (solver, H_batched, jump_ops)
+    args = (solver, rho0_batched, exp_ops, t_save)
+    kwargs = dict(H=H_batched, jump_ops=jump_ops)
```

Comment:
Similar comment here. It seems strange to separate `args` and `kwargs`, especially as the `kwargs` are not keyword arguments in the actual classes. I think you mean something like `generic_args` and `odeint_args`. These are not very good names but anything with a little bit more meaning would be nice I think !

Reply:
Ho but any argument following `*args` is by default a keyword argument (not to be confused with an argument having a default value!). I would argue the same that the generic Python naming are actually the easiest to understand.

Reply:
But I don't get why you would put `args` and `kwargs` on a different level here. They're both arguments of the qsolver classes. You might as well write `args = (solver, rho0_batched, exp_ops, t_save, H_batched, jump_ops)` no?

Reply:
No it wouldn't work, you can't have a positional argument following `*args`:
```python
>>> def foo(*args, a):
....    pass
>>> foo(1, 2, 3, 4)
TypeError: foo() missing 1 required keyword-only argument: 'a'
```

Reply:
Yes but it's because you changed the function signature from 
```python
def __init__(self, options: SolverOption, H: TDOperator, jump_ops: Tensor)
```
to 
```python
def __init__(self, *args, H: TDOperator, jump_ops: Tensor)
``` 
But ok I think you're right I'm just mixing myself up. It doesn't really matter anyway ^^

Reply:
Ha yes yes sorry, it's clearly because of the change 😅